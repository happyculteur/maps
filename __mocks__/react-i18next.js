const useMock = [k => k, {}];
useMock.t = k => k;
useMock.i18n = {};

module.exports = {
  useTranslation: jest
    .fn()
    .mockReturnValue(
      useMock
    )
};
