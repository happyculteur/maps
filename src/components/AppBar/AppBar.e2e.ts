/// <reference types="jest"/>
/// <reference types="cypress"/>

describe("AppBar component", () => {
  it("survey button shall redirect to survey", () => {
    cy.visit("/");

    cy.get('[data-e2e="SurveyButton"]').should("have.length", 1);
  });
});
