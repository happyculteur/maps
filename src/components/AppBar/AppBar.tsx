import { default as MuiAppBar } from "@material-ui/core/AppBar";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import Toolbar from "@material-ui/core/Toolbar";
import MenuIcon from "@material-ui/icons/Menu";
import { makeStyles } from "@material-ui/styles";
import * as React from "react";
import { useTranslation } from "react-i18next";

export const useStyles = makeStyles(theme => ({
  flex: {
    flex: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  root: {
    flexGrow: 1
  },
  surveyButton: {
    "&:hover": {
      backgroundColor: theme.palette.secondary.dark
    },
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.primary.light,
    marginLeft: "auto"
  }
}));

export const AppBar: React.SFC = props => {
  const classes = useStyles();
  const i18n = useTranslation();

  return (
    <div className={classes.root}>
      <MuiAppBar position="static" color="default">
        <Toolbar>
          <IconButton
            className={classes.menuButton}
            color="inherit"
            aria-label="Menu"
          >
            <MenuIcon />
          </IconButton>
          <Button
            className={classes.surveyButton}
            color="primary"
            href={process.env.REACT_APP_HAPPYCULTEUR_SURVEY}
            data-e2e="SurveyButton"
          >
            {i18n.t("Take the survey!")}
          </Button>
        </Toolbar>
      </MuiAppBar>
    </div>
  );
};

export default AppBar;
