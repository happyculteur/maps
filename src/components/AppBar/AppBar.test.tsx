import { shallow } from "enzyme";
import * as React from "react";
import { useTranslation } from "react-i18next";
import { AppBar, useStyles } from "./AppBar";

jest.mock("@material-ui/styles");
jest.mock("react-i18next");

describe("AppBar component", () => {
  describe("on shallow rendering", () => {
    it("should renders without crashing", () => {
      const wrapper = shallow(<AppBar />);

      expect(wrapper.debug()).toMatchSnapshot();
    });

    describe("should rely on hook", () => {
      it("useStyles", () => {
        shallow(<AppBar />);
        expect(useStyles).toHaveBeenCalledTimes(2);
      });
      it("useTranlation", () => {
        shallow(<AppBar />);
        expect(useTranslation).toHaveBeenCalledTimes(3);
      });
    });
  });
});
