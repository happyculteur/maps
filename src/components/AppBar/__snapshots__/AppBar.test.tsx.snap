// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component on shallow rendering should renders without crashing 1`] = `
"<div className={[undefined]}>
  <WithStyles(AppBar) position=\\"static\\" color=\\"default\\">
    <WithStyles(Toolbar)>
      <WithStyles(IconButton) className={[undefined]} color=\\"inherit\\" aria-label=\\"Menu\\">
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Button) className={[undefined]} color=\\"primary\\" href={[undefined]} data-e2e=\\"SurveyButton\\">
        Take the survey!
      </WithStyles(Button)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>"
`;
